### DISCUSSION TITLE
Illegible text in terminal

### DISCUSSION THREAD

**Discussion by goleary** [community/community#16871] [Codespaces] [answered]
I'm following the guide [here](https://docs.github.com/en/codespaces/getting-started/quickstart), but the text in the terminal is illegible

![image](https://user-images.githubusercontent.com/16123225/169118235-589df82b-7412-469e-b764-928636382f22.png)

Window 11
Brave V1.38.119

**Comment by DenverCoder1** [comment, marked as answer]
I found some solutions here: https://github.com/gitpod-io/gitpod/issues/5014

> Some browsers (eg. Brave) have privacy features that prevent some fingerprinting techniques that use the canvas such as checking what fonts are installed or determining how certain fonts are actual rasterized. [source](https://github.com/xtermjs/xterm.js/wiki/FAQ#characters-are-getting-cut-off-in-the-terminal)

### Option 1
Disable block fingerprinting at `brave://settings/shields` (https://github.com/gitpod-io/gitpod/issues/5014#issuecomment-1063944871)

### Option 2
Turn off shields for the site (it seems like this must be done for each project individually) (https://github.com/gitpod-io/gitpod/issues/5014#issuecomment-947238614)

### Option 3
Set `terminal.integrated.gpuAcceleration` to `canvas` in the settings (https://github.com/gitpod-io/gitpod/issues/5014#issuecomment-919901497)

```
"terminal.integrated.gpuAcceleration": "canvas"
```

### Option 4
Adjust the `terminal.integrated.lineHeight` and `terminal.integrated.letterSpacing` settings (https://github.com/gitpod-io/gitpod/issues/5014#issuecomment-891781810)

```
"terminal.integrated.lineHeight": 1.2,
"terminal.integrated.letterSpacing": 2
```

**Reply by dmilburn** [reply]
Hi from the Codespaces team, thanks for finding this solution! Would you and @goleary let us know if this fixes your problem?

**Reply by goleary** [reply]
Yup, this does fix the problem...at the expense of reduced privacy...doesn't look like there is any way to disable the fingerprinting block for a specific domain.

From the sounds of the discussion in the gitpod issue there doesn't seem to be any way around it though 😢 

Thanks!

**Reply by DenverCoder1** [reply]
Setting "Block fingerprinting" to "disabled" does fix the issue.

Just turning off shields for the specific page seems to work as well. This seems to only be possible per-subdomain, though (I can't find a way to turn off shields for all of `*.github.dev` at once).

An alternate solution that does not require reduced privacy would be to adjust the terminal line height and letter spacing in the `settings.json` as others [suggested in the issue](https://github.com/gitpod-io/gitpod/issues/5014#issuecomment-891781810) linked above. It's not exactly the same, but I tried it in Codespaces and it does also work in Codespaces.

Setting `"terminal.integrated.gpuAcceleration": "canvas"` also worked for me (as [suggested here](https://github.com/gitpod-io/gitpod/issues/5014#issuecomment-919901497))

**Reply by TechStudent10** [reply]
> Setting `"terminal.integrated.gpuAcceleration": "canvas"` also worked for me (as [suggested here](https://github.com/gitpod-io/gitpod/issues/5014#issuecomment-919901497))

Works for me on `Brave 1.38.119`
